phylum-dev/phylum-ci
# build: update `poetry` to latest version

Poetry is the workflow management tool used for this project and forms the root of all other actions taken when working with this repository. It is also used to manage dependencies and therefore should be treated very carefully, with updates to newer versions taken deliberately.

This PR updates `poetry` to the latest version of v1.7.0 to account for the [changes introduced](https://python-poetry.org/history) with these actions taken:

* Bump all instances of `poetry` to the new version
  * Installs in workflows
  * pre-commit hook revision
  * Dockerfiles
* Bump the `poetry-core` build system version in `pyproject.toml`
  * v1.8.0 is when `3.12` was added to list of available Python versions
  * v1.8.1 is just the latest version
  * This update is not strictly required to support Python 3.12
    * Bumping to ensure freshness
* Update documentation
  * Minimum version in `CONTRIBUTING.md`
* Update the lockfile with the new version of `poetry`
